qiskit/providers/backend.py

```diff
@@ -229,3 +240,435 @@ class can handle either situation.
+    def operations(self) -> List[Instruction]:
+        """A list of :class:`~qiskit.circuit.Instruction` instances that the backend supports."""
+        return list(self.target.operations)
+
+    def _compute_incomplete_basis(self):
+        incomplete_basis_gates = []
+        all_permutations_all_size = {}
+        for inst, qarg_props in self.target.items():
+            qargs = set(qarg_props)
+            size = len(next(iter(qargs)))
+            if size not in all_permutations_all_size:
+                all_permutations_all_size[size] = set(
+                    itertools.permutations(range(self.target.num_qubits), size)
+                )
+            if qargs != all_permutations_all_size[size]:
+                incomplete_basis_gates.append(inst)
+        self._basis_gates_all = incomplete_basis_gates
+
+    @property
+    def operation_names(self) -> List[str]:
+        """A list of instruction names that the backend supports."""
+        if self._basis_gates_all is None:
+            self._compute_incomplete_basis()
+        if self._basis_gates_all:
+            invalid_str = ",".join(self._basis_gates_all)
+            msg = (
+                f"This backend's operations: {invalid_str} only apply to a subset of "
+                "qubits. Using this property to get 'basis_gates' for the "
+                "transpiler may potentially create invalid output"
+            )
+            logger.warning(msg)
+        return list(self.target.operation_names)
+
+    @property
+    @abstractmethod
+    def target(self):
+        """A :class:`qiskit.transpiler.Target` object for the backend.
+
+        :rtype: Target
+        """
+        pass
+
+    @property
+    def num_qubits(self) -> int:
+        """Return the number of qubits the backend has."""
+        return self.target.num_qubits
+
+    @property
+    def coupling_map(self):
+        """Return the :class:`~qiskit.transpiler.CouplingMap` object"""
+        return self.target.coupling_map()
+
+    @property
+    def instruction_durations(self):
+        """Return the :class:`~qiskit.transpiler.InstructionDurations` object."""
+        return self.target.durations()
+
+    @property
+    @abstractmethod
+    def max_circuits(self):
+        """The maximum number of circuits (or Pulse schedules) that can be
+        run in a single job.
+
+        If there is no limit this will return None
+        """
+        pass
+
+    @classmethod
+    @abstractmethod
+    def _default_options(cls):
+        """Return the default options
+
+        This method will return a :class:`qiskit.providers.Options`
+        subclass object that will be used for the default options. These
+        should be the default parameters to use for the options of the
+        backend.
+
+        Returns:
+            qiskit.providers.Options: A options object with
+                default values set
+        """
+        pass
+
+    def t1(self, qubits: Union[int, List[int]]) -> Union[float, np.array]:
```

Comment:
Is this still required? I'm fine with only having `backend.qubit_properties(0).t1`.

Reply:
It's not required, I just left it in for a shorthand so @nonhermitian could do `backend.t1(range(backend.num_qubits)).mean()` to get the average T1 for a device. But `np.array([backend.qubit_properties(i).t1 for i in range(num_qubits)]).mean()` does work just as well I guess.

Reply:
This seems specific to a type of backend (like IBMQ backend) rather than something for base class

Reply:
Removed in: https://github.com/Qiskit/qiskit-terra/pull/5885/commits/d2c279659e1be655388a45225f359c09b081f6ed